bndtools.core/src/bndtools/central/Central.java

```diff
@@ -228,6 +237,7 @@ public static Workspace getWorkspace() throws Exception {
                 if (ws != null) {
                     ws.close();
                 }
+                workspaceQueue.fail(e); // notify onWorkspaceInit callbacks
```

Comment:
You cannot resolve the deferred in the synchronized block since the callback will occur holding the lock.